snapcraft.yaml

```diff
@@ -101,27 +101,27 @@ parts:
       - usr/share/doc/libmm-glib0
 
   modemmanager:
-    plugin: autotools
+    plugin: meson
     after: [ stack-snaps-tools ]
 
     source: https://git.launchpad.net/ubuntu/+source/modemmanager
     source-type: git
-    source-branch: applied/ubuntu/jammy-updates
+    source-branch: applied/ubuntu/noble
 
-    autotools-configure-parameters:
+    meson-parameters:
       - --prefix=/usr
       - --libdir=/usr/lib
       - --libexecdir=/usr/lib/ModemManager
       - --sysconfdir=/etc
-      - --enable-gtk-doc=no
-      - --with-polkit=no
-      - --enable-vala=no
-      - --disable-maintainer-mode
-      # Set explicitly CFLAGS until lp: #1791946 is solved
-      - CFLAGS='-O2 -Wl,-z,relro,-z,now -pie -fpie'
+      - -Dgtk_doc=false
+      - -Dpolkit=no
+      - -Dvapi=false
+    
+    build-environment:
+      - CFLAGS: "-O2 -Wl,-z,relro,-z,now"
```

Comment:
We were asked to enable PIE for security hardening reasons (to support ASLR). Maybe this is now the default, but please make sure that it is actually enabled.

Reply:
Jesus that is annoying, it's conflicting with other build options, let me find a different approach then

Reply:
Found the correct meson switch for this - updated!

Reply:
Double checking on this, are we sure we are using -O2 as well?

Reply:
Yes, -O2 is present when compiling